lib/fluent/plugin/in_s3.rb

```diff
@@ -181,6 +183,11 @@ def run
         @poller.poll(options) do |message|
           begin
             body = Yajl.load(message.body)
+            if @assume_sns
+              log.debug(body)
```

Comment:
how about putting the following after L191? it can handle both SNS messages and other normal messages.

```rb
if @assume_sns && body['Message']
  body = Yajl.load(body['Message'])
end
```

Reply:
Hmm, I guess? Seems like an optimization for an impossible case, where a single bucket is sending events via both SNS and SQS. This would be an extremely unusual situation I think?

Reply:
Ah, You're right. it's okay as is.